src/opentelemetry-tracing.ts

```diff
@@ -31,13 +38,21 @@ export class OpenTelemetryTracer {
    */
   createSpan(
     spanName: string,
-    attributes?: Attributes,
+    attributes?: SpanAttributes,
     parent?: SpanContext
   ): Span {
     const tracerProvider: Tracer = trace.getTracer('default') as Tracer;
-    return tracerProvider.startSpan(spanName, {
-      parent: parent,
-      attributes: attributes,
-    });
+
+    let spanContext = undefined;
+    if (parent) {
+      spanContext = setSpanContext(context.active(), parent);
+    }
```

Comment:
I'd recommend just setting parent in context immediately when you extract it from the message. Then you don't need to pass it here, it should be implicit.

Looks like this whole class could just be a function as well, if you have time for a little refactor 😄

Reply:
I have applied your feedback, let me know if this is what you had in mind :)

Reply:
There is one test which even in `master` is incorrect but I can't seem to make it work either so not sure what to do. It's related to the subscriber's test named: 'exports a span once it is created' which asserts `assert.ok(spans)` which is `true` when `exporter.getFinishedSpans()` an empty array and its not explicitly checking if at least one span is returned.

If you or anyone else knows how to fix this, that would be great! I am not seeing it.

Reply:
This test has been fixed and passing